model/map-ldap/src/main/java/org/keycloak/models/map/storage/ldap/role/entity/LdapRoleEntity.java

```diff
@@ -0,0 +1,341 @@
+/*
+ * Copyright 2022. Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.models.map.storage.ldap.role.entity;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.EnumMap;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import org.apache.commons.lang.NotImplementedException;
+import org.keycloak.models.ModelException;
+import org.keycloak.models.map.common.DeepCloner;
+import org.keycloak.models.map.common.EntityField;
+import org.keycloak.models.map.common.UpdatableEntity;
+import org.keycloak.models.map.common.delegate.EntityFieldDelegate;
+import org.keycloak.models.map.role.MapRoleEntity;
+import org.keycloak.models.map.role.MapRoleEntityFields;
+import org.keycloak.models.map.storage.ldap.model.LdapMapDn;
+import org.keycloak.models.map.storage.ldap.model.LdapMapObject;
+import org.keycloak.models.map.storage.ldap.role.config.LdapMapRoleMapperConfig;
+import org.keycloak.models.map.storage.ldap.role.LdapRoleMapKeycloakTransaction;
+
+public class LdapRoleEntity extends UpdatableEntity.Impl implements EntityFieldDelegate<MapRoleEntity> {
+
+    private final LdapMapObject ldapMapObject;
+    private final LdapMapRoleMapperConfig roleMapperConfig;
+    private final LdapRoleMapKeycloakTransaction transaction;
+    private final String clientId;
+    // the realmId might not be known at retrieval time when retrieved by ID. will be set later by the MapRoleProvider
+    private String realmId;
+
+    private static final EnumMap<MapRoleEntityFields, BiConsumer<LdapRoleEntity, Object>> setters = new EnumMap<>(MapRoleEntityFields.class);
```

Comment:
Nit: use upper case for `static` fields (applies to other fields within this file as well)

Reply:
fixed.